src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedValuesCodeFixProvider.cs

```diff
@@ -126,18 +126,35 @@ public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
                         {
                             // Do not offer a fix to replace unused foreach iteration variable with discard.
                             // User should probably replace it with a for loop based on the collection length.
-                            return Task.CompletedTask;
+                            return;
                         }
 
                         title = FeaturesResources.Use_discard_underscore;
+
+                        // Check if this is compound assignment which is not parented by an expression statement,
+                        // for example "return x += M();" OR "=> x ??= new C();"
+                        // If so, we will be replacing this compound assignment with the underlying binary operation.
+                        // For the above examples, it will be "return x + M();" AND "=> x ?? new C();" respectively.
+                        // For these cases, we want to show the title as "Remove redundant assignment" instead of "Use discard _".
+
+                        var syntaxFacts = context.Document.GetLanguageService<ISyntaxFactsService>();
+                        var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+                        var node = root.FindNode(context.Span, getInnermostNodeForTie: true);
+                        if (syntaxFacts.IsLeftSideOfAnyAssignment(node) &&
+                            !syntaxFacts.IsLeftSideOfAssignment(node) &&
```

Comment:
i can't make sense of this check... how can a node both be the left side of any assignment... but not hte left side of an assignment?

Reply:
IsLeftSideOfAnyAssignment eventually comes down to this check: http://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp/Syntax/SyntaxKindFacts.cs,fed59b31290ca8e2

IsLeftSideOfAssignment checks if it is a simple assignment expression.

Reply:
i guess i'm not understanding then.  why does this only aply to compound assignment?  if you had `x = new C()` wouldn't you want to remove this redundant assignment just as muhc as `x ??= new C()`?

Reply:
For redundant assignment `x = new C()`, code fix changes it to `_ = new C()`, and the current code fix title of `Use discard _` is appropriate.

For redundant compound assignment `x ??= new C()`, code fix changes it to `x ?? new C()`, so the current code fix title of `Use discard _` is misleading. This change adjust this case for a more suitable code fix title.

Reply:
Note the code fix itself is working fine, as noted in https://github.com/dotnet/roslyn/issues/38507. The title is misleading for compound assignments.

Reply:
i see.  could you instead use `IsCompoundAssignExpression`?  i think that will be much easier to understand :)

Reply:
Note: you might need to update that helper to add CoalesceAssignmentExpression.  Alternatively, you could write it to do node+kind checks to be safe when new assignment exprs are added.

Reply:
> you might need to update that helper to add CoalesceAssignmentExpression

I was worried what might break if I change that helper to add CoalesceAssignmentExpression :-). Let me do so and find out.

Reply:
the answer should ideally be 'nothing'. t he purpose of the helper is for features to say: i want to continue to operate on new compound assignments as they get added :)